x/poe/contract/tg4_engagement.go

```diff
@@ -53,6 +54,11 @@ type WithdrawRewardsMsg struct {
 	Receiver *string `json:"receiver,omitempty"`
 }
 
+// DelegateWithdrawalMsg contract sets given address as allowed for senders funds withdrawal
+type DelegateWithdrawalMsg struct {
+	Delegate string `json:"delegate"`
```

Comment:
```suggestion
	Delegate string `json:"delegated"`
```
The use of the past participle in the contract here is a poor choice. This could perhaps have been named "address", for simplicity and clarity.

You might consider changing the field name as well, for clarity.